rust/kernel/error.rs

```diff
@@ -57,14 +57,34 @@ impl Error {
     pub const EBADF: Self = Error(-(bindings::EBADF as i32));
 
     /// Creates an [`Error`] from a kernel error code.
+    /// when errno given is invalid,
+    /// it will print a warning message and convert it to EINVAL
     pub fn from_kernel_errno(errno: c_types::c_int) -> Error {
+        Error::new(errno)
+    }
+
+    /// Creates an [`Error`] from a kernel error code without a sanity check
+    pub fn from_kernel_errno_unchecked(errno: c_types::c_int) -> Error {
         Error(errno)
     }
 
     /// Returns the kernel error code.
     pub fn to_kernel_errno(self) -> c_types::c_int {
         self.0
     }
+
+    /// INVARIANT: make sure Error is initialized with a sane value
+    /// When an invalid errno is found, it will
+    /// 1) print a warning message
+    /// 2) convert it to EINVAL
+    pub fn new(errno: c_types::c_int) -> Error {
+        if errno < -(bindings::MAX_ERRNO as i32) || errno >= 0 {
+            crate::pr_warn!("Creating Error with an invalid errno {}, convert it to EINVAL", errno);
+            Error::EINVAL
```

Comment:
If you use `return` here, then you can remove the `else` and its indentation level; i.e. try to keep the happy path unindented whenever possible.